iothub/device/src/Transport/ProtocolRoutingDelegatingHandler.cs

```diff
@@ -118,7 +118,7 @@ public override async Task WaitForTransportClosedAsync()
             Debug.Assert(InnerHandler != null);
 
             // We don't need to double check since it's being handled in OpenAsync
-            CreateNewTransportHandler();
+            CreateNewTransportIfNotReady();
```

Comment:
This is breaking the spec: the transport needs to be disposed if it reported disconnect.

Reply:
Offline discussion. (Design spec to follow )

Reply:
Table P0 (Major fix)